modules/common/src/main/Bus.scala

```diff
@@ -23,23 +15,25 @@ object NamedBus:
     import lila.core.timeline.*
     def apply(propagate: Propagate): Unit = Bus.publish(propagate, "timeline")
 
-final class BusChannel(channel: Bus.Channel):
-  def apply(msg: Bus.Payload): Unit                       = Bus.publish(msg, channel)
+final class BusChannel(channel: Channel):
+  def apply(msg: Payload): Unit                           = Bus.publish(msg, channel)
   def subscribe(subscriber: Bus.SubscriberFunction): Unit = Bus.subscribeFun(channel)(subscriber)
 
 object BusChannel:
   val forumPost = BusChannel("forumPost")
 
 object Bus:
 
-  type Channel    = String
-  type Subscriber = Tellable
-  type Payload    = Matchable
-
   type SubscriberFunction = PartialFunction[Payload, Unit]
 
   val named = NamedBus
-  val chan  = BusChannel
+
+  def pub[T <: Payload](payload: T)(using wc: WithChannel[T]) =
+    publish(payload, wc.channel)
+  def sub[T <: Payload: Typeable](f: PartialFunction[T, Unit])(using
+      wc: WithChannel[T]
+  ) =
+    subscribeFun(wc.channel)({ case x: T => f(x) })
```

Comment:
```suggestion
    subscribeFun(wc.channel):
      case x: T => f(x)
```
If you hate braces :D